src/main/resources/api/service_openapi.yaml

```diff
@@ -1301,14 +1301,29 @@ components:
           $ref: '#/components/schemas/PrivateResourceUser'
 
     PrivateResourceUser:
-      description: A user and role pair for a user of a private resource.
+      description: User and role information for a user of a private resource.
       type: object
       properties:
         userName:
           description: Email address for the user of the resource
           type: string
-        iamRole:
-          $ref: '#/components/schemas/IamRole'
+        privateResourceIamRole:
+          $ref: '#/components/schemas/PrivateResourceIamRole'
+
+    PrivateResourceIamRole:
+      description: Description of role(s) granted to the user of a private resource
+      type: object
+      required: [role, editor]
+      properties:
+        role:
+          description: What access role (reader or writer) this user has
+          type: string
+          enum: ['READER', 'WRITER']
+        editor:
```

Comment:
nit: This seems a bit ad hoc, but it should work. The business rule that says readers can't be writers but they can be editors seems like something I wouldn't bake into the API. So I guess  it's fine if this is a fundamental abstraction/rule.

Reply:
At least currently this is how we've defined private resource roles - either reader or writer (which also includes `read` permission), and independently possibly editor.

Reply:
Another alternative is to accept a list of roles. If both READER and WRITER are specified, we choose writer. That would leave us a plain enumeration of possible roles. Seems more extensible to me.

Reply:
Went with the list of roles approach here for extensibility